libs/csm/extensions/csm_ext.py

```diff
@@ -107,9 +112,66 @@ def destroy_prep_deploy_cluster(self, expect_fail=None):
             self.log.info("[Start] Sleep %s", self.update_seconds)
             time.sleep(self.update_seconds)
             self.log.info("[End] Sleep %s", self.update_seconds)
-         
+
             self.log.info("Cleanup: Check cluster status")
             resp = self.ha_obj.poll_cluster_status(self.master)
             assert resp[0], resp[1]
             self.log.info("Cleanup: Cluster status checked successfully")
         self.log.info("[End] Redeploy the setup...")
+
+    def delete_data_pod(self):
+        """
+        Degrade cluster safely by deleting a data pod
+        """
+        self.log.info("[Start] Degrade the cluster...")
+        result = False
+        self.log.info("Get pod to be deleted")
+        sts_dict = self.master.get_sts_pods(pod_prefix=POD_NAME_PREFIX)
+        sts_list = list(sts_dict.keys())
+        self.log.debug("%s Statefulset: %s", POD_NAME_PREFIX, sts_list)
+        sts = random.sample(sts_list, 1)[0]
+        delete_pod = sts_dict[sts][-1]
+        self.log.info("Pod to be deleted is %s", delete_pod)
+        set_type, set_name = self.master.get_set_type_name(pod_name=delete_pod)
+        resp = self.master.get_num_replicas(set_type, set_name)
+        assert resp[0], resp
+        num_replica = int(resp[1])
+        num_replica-=1
+        self.log.info("Shutdown data pod by replica method and "
+                    "verify cluster & remaining pods status")
+        resp = self.ha_obj.delete_kpod_with_shutdown_methods(
+            master_node_obj=self.master, health_obj=self.hlth_master,
+            delete_pod=[delete_pod], num_replica=num_replica)
+        pod_name = list(resp[1].keys())[0]
+        set_name = resp[1][pod_name]['deployment_name']
+        self.restore_method = resp[1][pod_name]['method']
+        if resp[1] is None:
+            self.log.error("Failed to shutdown/delete pod: %s", resp)
+        if resp[0] is not True:
+            self.log.error("Cluster/Services status is not as expected: %s", resp)
+        else:
+            result = True
+        self.log.info("Successfully shutdown data pod %s. Verified cluster and "
+                    "services states are as expected & remaining pods status is online.", pod_name)
+        self.log.info("[End] Degrade the cluster...")
+        return result,set_name,num_replica
+
+    def restore_data_pod(self, set_name:str, num_replica:int):
+        """
+        Restore cluster safely by deleting a data pod
```

Comment:
Is this correct comment? - we are not deleting pod here

Reply:
updated